Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/submission-result-details/SubmissionResultsDetails.tsx

```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import { BiCaretDown, BiCaretUp } from 'react-icons/bi';
+
+import { ITestRunDetailsType } from '../../../../hooks/submissions/types';
+import { useHttp } from '../../../../hooks/use-http';
+import { getUserAuthInfoUrl } from '../../../../utils/urls';
+import IconSize from '../../../guidelines/icons/common/icon-sizes';
+import Icon from '../../../guidelines/icons/Icon';
+import TestRunDiffView from '../../test-run-diff-view/TestRunDiffView';
+
+import styles from './SubmissionResultsDetails.module.scss';
+
+interface ITestCaseRun {
+    id: number;
+    checkerComment?: string;
+    executionComment?: string;
+    expectedOutputFragment?: string;
+    input?: string;
+    isTrialTest: boolean;
+    memoryUsed: number;
+    orderBy: number;
+    resultType: string;
+    showInput: boolean;
+    submissionId?: number;
+    timeUsed: number;
+    userOutputFragment?: string;
+}
+
+interface ITestRunDetailsCollapsed {
+    [id: string]: {
+        isExpanded: boolean;
+        detailsExpanded: boolean;
+    };
+}
+interface ISubmissionResultsDetails {
+    testRuns?: ITestCaseRun[];
+}
+
+interface IUserRole {
+    id: string;
+    name: string;
+}
+
+interface IUserAuthData {
+    email: string;
+    id: string;
+    roles: IUserRole[];
+    userName: string;
+}
+
+const SubmissionResultsDetails = ({ testRuns }: ISubmissionResultsDetails) => {
+    const [ isUserAdmin, setIsUserAdmin ] = React.useState<boolean>(false);
+    const [ testRunDetailsCollapsed, setTestRunDetailsCollapsed ] = React.useState<ITestRunDetailsCollapsed>({});
+
+    const { get, data } = useHttp<null, IUserAuthData>({ url: getUserAuthInfoUrl() });
+
+    React.useEffect(() => {
+        get();
+    }, [ get ]);
+
+    React.useEffect(() => {
+        const { roles } = data || {};
+        const isUserRoleAdmin = roles?.find((role: IUserRole) => role.name === 'Administrator');
+
+        if (!isUserRoleAdmin) {
+            return setIsUserAdmin(false);
+        }
+
+        return setIsUserAdmin(true);
+    }, [ data ]);
+
+    const renderTestHeading = (testRun: ITestCaseRun, idx: number) => {
+        const isWrongAnswer = testRun.resultType === 'WrongAnswer' || testRun.resultType === 'RunTimeError';
+        let testRunText = `Test #${testRun.orderBy}`;
+        if (testRun.isTrialTest) {
+            testRunText = `Zero ${testRunText}`;
+        }
+
+        if (isWrongAnswer) {
+            if (testRun.resultType === 'RunTimeError') {
+                testRunText += ' (Compile time Error)';
+            } else {
+                testRunText += ' (Incorrect Answer)';
+            }
+        } else {
+            testRunText += ' (Correct Answer)';
+        }
+
+        return (
+            <h2
+              id={`test-heading-${idx}`}
+              style={{
+                  fontSize: 30,
+                  margin: '10px 0',
```

Comment:
Move to scss files

Reply:
Here I kept it inline, since we have conditional style as well. I can move the other styles in a classname and keep the color inline, since its conditional

Reply:
We use concatClassnames when conditional styling is present. Example:

```
const className = concatClassNames(
        container,
        official
            ? officialBackground
            : '',
    );
```